src/verifier.js

```diff
@@ -172,9 +172,11 @@ function verifyToken(
   validateAlgorithmAndSignature(input, header, signature, key, allowedAlgorithms)
 
   // Verify typ
-  const headerTyp = (header.typ || '').toLowerCase().replace(/^application\//, '')
-  if (checkTyp && (checkTyp !== headerTyp)) {
-    throw new TokenError(TokenError.codes.invalidType, 'Invalid typ.')
+  if (checkTyp) {
+    const headerTyp = (header.typ || '').toLowerCase().replace(/^application\//, '')
```

Comment:
```suggestion
    const headerTyp = (header.typ.toLowerCase().replace(/^application\//, '') || '')
```

Reply:
I think this will throw if `herader.typ` is not defined.

Reply:
fair enough. but is there a point to call `.toLowerCase().replace(/^application\//, '')` when it's a defaulted to `''` value?

Also, the whole thing will throw when header.typ is of an incorrect type anyway. I don't see a Header Parameter or Claim type checks anywhere.

Reply:
Removing the `|| ''` will cause this function to throw, which would mean it would throw a different error code.